dashboard/src/components/RevisionCard.tsx

```diff
@@ -0,0 +1,141 @@
+ * @param {Date} revisionDate - the revision date
+ * @param {string} previousVersion - the previous version
+ * @param {string} currentVersion - the current version
+ * @param {string} statusCode - the status code
+ *
+ * @returns {JSX.Element} - the RevisionCard component
+ *
+ * TODO: implement correct comparison of version strings.
+ *
+ */
+
+import React from "react";
+import "../index.css";
+import Status from "./Status";
+export default function RevisionCard({
+  revision,
+  revisionDate,
+  previousVersion,
+  currentVersion,
+  statusCode,
+  isActive = "false",
+  isRefreshable = false,
+  onClick
+}: {
+  revision: string;
+  revisionDate: Date;
+  previousVersion: string;
+  currentVersion: string;
+  statusCode: "Deployed" | "Superseded" | "Failed";
+  isActive: "true" | "false";
+  isRefreshable: boolean;
+  onClick: () => void;
+}): JSX.Element {
+  const activeVariants = {
+    true: "border-1 border-blue-500",
+    false: "border-1 border-gray-300"
+  };
+  return (
+    <div
+      className={`card ${activeVariants[isActive]} relative h-24 w-64 hover:card-hover m-1`}
+      onClick={onClick}
+    >
+      {/* the status is displayed in the top left corner, using the Status component */}
+      <div className="absolute top-0 left-0 mt-4 ml-4">
+        <Status isRefreshable={isRefreshable} statusCode={statusCode} />
+      </div>
+
+      {/* the revision number is displayed in the top right corner */}
+      <div className="absolute top-0 right-0 mt-4 mr-4">
+        <span className="text-sm font-bold">#{revision}</span>
+      </div>
+      {/* the version-change sub part is displayed in the bottom left corner */}
+      <div className="absolute bottom-0 left-0 mb-4 ml-4">
+        <span className="text-sm font-light">{previousVersion}</span>
+        {/* the correct symbol is displayed based on the previous and current */}
+        {/* automatically, and the symbol is displayed directly */}
+        {/* it's an up-arrow if the current version is higher than the previous, */}
+        {/* and a down-arrow if the current version is lower than the previous */}
+        {currentVersion > previousVersion ? (
+          <span className="text-sm font-light"> &#8593; </span>
+        ) : (
+          <span className="text-sm font-light"> &#8595; </span>
+        )}
+        <span className="text-sm font-light">{currentVersion}</span>
+      </div>
+
+      {/* if the revision was made longer than 60 seconds ago, the time is displayed in minutes */}
+      {/* if the revision was made longer than 60 minutes ago, the time is displayed in hours */}
+      {/* if the revision was made longer than 24 hours ago, the time is displayed in days */}
+      <div className="absolute bottom-0 right-0 mb-4 mr-4">
+        <span className="text-sm font-light">
+          AGE:{" "}
+          {Math.floor((Date.now() - revisionDate.getTime()) / 1000) < 60
+            ? Math.floor((Date.now() - revisionDate.getTime()) / 1000) + "s"
```

Comment:
extract to a function describing what is happening here.

Reply:
Extracted into getRevisionAgeJSX